docs/developer-guide/working-with-custom-parsers.md

```diff
@@ -0,0 +1,72 @@
+# Working with Custom Parsers
+
+If you want to use your own parser and provide additional capabilities for your rules, you can specify your own custom parser. If a `parseForESLint` method is exposed on the parser, this method will be used to parse the code. Otherwise, the `parse` method will be used. Both methods should take in the source code as the first argument, and an optional configuration object as the second argument (provided as `parserOptions` in a config file). The `parse` method should simply return the AST. The `parseForESLint` method should return an object that contains the required property `ast` and an optional properties `services`, `scopeManager`, and `visitorKeys`.
```

Comment:
I know you didn't introduce this, but since we're moving the docs, let's clean them up a bit.

In the following segment:

```
The `parseForESLint` method should return an object that contains the required property `ast` and an optional properties `services`, `scopeManager`, and `visitorKeys`.
```

Can you please remove the "an" before "optional properties"? Thanks!